In [5]:
import numpy as np
from scipy import optimize
from scipy.optimize import fsolve


**Application of the extended real associated solution (ERAS) model to excess molar volumes of binary mixtures of (alkoxyethanols + 1-alkanol)**

Amalendu Pala, Harsh Kumarb, Shanti Lal Oswalc

a Department of Chemistry, Kurukshetra University, Kurukshetra-136 119, Haryana, India

b Department of Chemistry, Dr B R Ambedkar National Institute of Technology, Jalandhar-144 011 Punjab, India

c R & D, Biochemistry Division, Span Diagnostic Limited, 173-B, New Industrial Estate, Udhana, Surat-394 120, Gujarat, India

Received 12 February 2010, Revised 23 June 2010, Accepted 9 July 2010, Available online 17 July 2010

http://www.sciencedirect.com/science/article/pii/S0167732210002199

**Modelo ERAS**

El modelo ERAS se utiliza para calcular propiedades termodinámicas:

1. Entalpía molar de exceso $H^E_m$
2. Volumen molar de exceso $V^E_m$

de sustancías puras y mezclas binarias.

As per ERAS model, any excess function $Z^E$ is the combination of the two additive terms so called chemical contribution ($Z^E_{chem}$) and physical contribution ($Z^E_{phy}$) which arise from hydrogen bonding effects and non-polar Vander Waals interactions between unlike molecules including free volume effect, respectively and is represented as:

$$  Z^E = Z^E_{chem} + Z^E_{phy} $$

where ZE = excess molar volume, VE or excess molar enthalpy, HE.
The assumption made in the frame of ERAS model that only consecutive association occurs, which is described by a chemical equilibrium constant KA, according to the equations:

The assumption made in the frame of ERAS model that only consecutive association occurs, which is described by a chemical equilibrium constant KA, according to the equations:

\begin{equation}
A_m + A \rightleftarrows^{K_A} A_{m + 1}
\end{equation}

$$ B_m + B \rightleftarrows^{K_B} B_{m + 1} $$

where m and n ranges from 1 to ∞, are the degree of self association. The cross-association between two self associated components A and B is represented as:

$$ A_m + B_n \rightleftarrows^{K_{AB}} A_mB_n $$





The association constant $K_i$ (i = A, B and AB) are assumed to be independent of the chain length. Their temperature dependence is given by:

$$K_i = K_0exp\left(- \frac{\Delta h^*_i}{R} \left( \frac{1}{T} - \frac{1}{T_0}\right)\right) $$


where $K_0$ is the equilibrium constant at standard temperature $T_0$ (298.15 K), R is the gas constant and $\Delta h^*_i$ is the enthalpy for the reactions given by the Eqs.  (2), (3) and (4), which corresponds to the hydrogen bonding energy. These reactions are also characterized by the volume change $\Delta v^*_i$ related to formation of linear chains.
















The expression for $V_E$ of the ERAS model extended to the two block approach of cross-association [64] are given by [19] and [20]

$$ V^E_{ERAS} = V^E_{phy} + V^E_{chem} $$

$$ V^E_{phy} = \left(x_AV^*_A + x_BV^*_B\right) \left(\hat V_M - \phi_A \hat V_A - \phi_B \hat V_B \right) $$

$$ V^E_{chem} = \hat V_M \left( x_AK_A \Delta v^*_A  \left(\phi_{A1}  - \phi_{A0} \right) + x_BK_B \Delta v^*_B  \left(\phi_{B1}  - \phi_{B0} \right) + x_AK_{AB} \Delta v^*_{AB} \left( \frac{\phi_{B1} \left(1 - K_A\phi_{A1}\right)} {\frac{V_B}{V_A} \left(1 - K_B\phi_{B1}\right) + K_{AB}\phi_{B1}}\right) \right)$$








$K_A$ and $K_B$ are the equilibrium constants. $K_{AB}$ and $\Delta h^*_{AB}$ are the association constants and hydrogen bond energy from the cross-association. $\phi_{A1}$ and $\phi_{B1}$ are the hard core volume fraction of the component in the mixture. They have to be calculated numerically from the solution of the following coupled equations:

$$ \phi_A = \frac{ \phi_{A1} } { \left( 1 - K_A \phi_{A1}\right)^2} \left(1 + \frac{ V_A K_{AB} \phi_{AB1} } {V_B \left(1- K_B \phi_{B1} \right) } \right) $$


$$ \phi_B = \frac{ \phi_{B1} } { \left( 1 - K_B \phi_{B1}\right)^2} \left(1 + \frac{ K_{AB} \phi_{A1} } {\left(1- K_A \phi_{A1} \right) } \right) $$

The physical contribution $Z^E_{phy}$ in ERAS model is derive from Flory's equation of state [5] and [6], which is assumed to be valid not only for pure components but also for the mixture.


$$ \frac {P_i \hat V_i} {\hat T_i} = \frac{\hat V_i^{4/3}} {\hat V_i^{1/3} - 1} - \frac{1} {\hat V_i \hat T_i} $$

where i = A, B, M (mixture).

All the reduction parameters of pure components can be determined knowing the experimental data for molar volume V, thermal expansion coefficient $\alpha$, isothermal compressibility $k_T$, provided suitable association parameters, $K_i$, $\Delta v^*_i$, $\Delta h^*_i$ are known. The reduction parameters for the mixtures $P^*_i$, $T^*_i$ and $V^*_i$ are calculated from mixing rules [19], [20] and [25]as follows

$$ P^*_M =  \phi_A P^*_A + \phi_B P^*_B - X_{AB}\phi_A \theta_B $$

$$ T^*_M =  \frac{P^*_M} {\frac{\phi_A P^*_A}{T^*_M } + \frac{\phi_B P^*_B}{T^*_M } } $$

$$ V^*_M =  x_A V^*_A + x_B V^*_B $$

$X_{AB}$ is an interaction parameter characterizing the difference of dispersive intermolecular interaction between A and B. $\phi$ i and $\theta_i$ are the hard-core volume fraction and the surface fraction of the component i [8].

**Parámetros de sustancias puras**

**Parámetros caracteristicos**

En caso de que los componentes no presenten asociación entre sus propias moléculas se utiliza la teoría de Flory directamente, para $K_i = 0$

$$ P^*_i = \frac{\alpha}{k_T}T \hat V_i^2$$

$$ V^*_i = \frac{V_i}{\hat V_i}  $$

$$ T^*_i = \frac{T}{\hat T_i}  $$

$$ \hat T_i = T \frac{\hat V_i^{4/3}} {\hat V_i^{1/3} - 1}  $$

**falta corroborar la siguiente ecuación:**

$$ \hat V^{\frac{1}{3}} - 1 = \frac {\frac{\alpha T}{3}} {1 + \alpha T}$$


En caso de tener una asociación entre moléculas del mismo componente, se utiliza las ecuaciones:


$$ V^*_i = V_i \left(\frac{1+(\alpha_i - \alpha*_i)T}{1+ 4/3(\alpha_i - \alpha*_i)T} \right)^{3} $$



$$ \alpha^* = \Delta v^*_i \Delta h^*_i \frac{(4K_i+1)^{1/2}-2K_i(4K_i+1)^{-1/2}-1}{2K_iV^*_iRT^2}  $$ 



$$ P^*_i = \left(\alpha_i - \alpha^*_i  \right) T\hat V^2_i \left(k_{T_{i}}-̈́\alpha^*_iT \frac{\Delta v_i}{\Delta h_i} \right)^{-1}$$



$$ T^*_i = \frac{T}{\hat T_i}  $$



$$ \hat T_i = T \frac{\hat V_i^{4/3}} {\hat V_i^{1/3} - 1}  $$

En este caso las funciones que predicen los resultados de $V^*_i$ y $\alpha^*_i$ del 1-Heptanol en la Tabla 6 del artículo: *Experimental study and ERAS modeling of the excess molar enthalpy of (acetonitrile + 1-heptanol or 1-octanol) mixtures at (298.15, 313.15, and 323.15) K and atmospheric pressure*, son:

http://www.sciencedirect.com/science/article/pii/S0021961408000207

$$ f_1 = 1 - \frac{V_i}{V^*_i} \left(\frac{1+(\alpha_i - \alpha*_i)T}{1+ \frac{4}{3}(\alpha_i - \alpha*_i)T} \right)^{3} $$


$$ f_2 = \alpha^*_i - \Delta v^*_i \Delta h^*_i \frac{(4K_i+1)^{1/2}-2K_i(4K_i+1)^{-1/2}-1}{2K_iV^*_iRT^2}  $$ 


In [6]:
def volumen_cal(x):
    
    """
    
    T [=] K
    R [=] 8.314472e-3 kJ/(K * mol)
    V [=] cm^3 / mol
    alfa [=] 1 / K
    Ah [=] kJ / mol
    Av [=] cm^3 / mol
    Vs [=] adimensional
    K1 [=] adimensional
    
    """
    
    V_g = np.array(x[0])   
    alfa_p = np.array(x[1])    
    funcion_1 = 1 - V_g * ((1 + (alfa1 - alfa_p)*T)/(1 + 4/3 * (alfa1 - alfa_p)*T))**3
    constante_1 = (4*K1+1)**(0.5) - 2*K1*(4*K1+1)**(-0.5) - 1
    #funcion_2 = 1 - Av1 * Ah1 * (constante_1*V_g) / (2 * K1 * R * T**2 * V1*alfa_p)
    funcion_2 = alfa_p - Av1 * Ah1 * (constante_1*V_g) / (2 * K1 * R * T**2 * V1)
    
    f = [funcion_1, funcion_2]
    
    #print (f)
    return f

d = 3

if d == 1:
    #------------------------------------------------------------
    K1 = 69.04    
    alfa1 = 7.60e-4   
    kT1 = 8.20e5
    V1 = 141.80
    Av1 = -5.60
    Ah1 = -18.99
    
    T = 298.15
    R = 8.314472e-3
elif d == 2:
    #------------------------------------------------------------
    K1 =  47.83   
    alfa1 = 9.20e-4   
    kT1 = 8.30e5
    V1 = 143.69
    Av1 = -5.60
    Ah1 = -18.99
    
    T = 313.15
    R = 8.314472e-3
    #------------------------------------------------------------
elif d == 3:
    K1 = 38.17
    alfa1 = 9.90e-4
    kT1 = 8.40e5
    V1 = 144.95
    Av1 = -5.60
    Ah1 = -18.99
    #------------------------------------------------------------
    
    T = 323.15
    R = 8.314472e-3

#sol = optimize.root(volumen_cal, [10, 100], method='hybr')
#sol = fsolve(volumen_cal,[10, 100],args=(initial_pressure, 1, 2, Avsl), xtol=1e-4)
sol = fsolve(volumen_cal,[10, 100], xtol=1e-5)
sol

print ("V_g = {0}".format(sol[0]))
print ("alpha*_i = {0}".format(sol[1]))

V_g = sol[0]
V_p = V1/V_g
print("V* = {0}".format(V_p))

alfa_p = sol[1]

#alfa_p = 7.2297e-5

kT1 = kT1 * 1e6

print((kT1 - alfa_p * T * (Av1/Ah1)))

P1_p = ((alfa1 - alfa_p) * T * V_g ** 2) / ((kT1 - alfa_p * T * (Av1/Ah1)))
P1_p = P1_p * 1e-6
print("P* = {0}".format(P1_p))

T1_p = T * (V_g**(4/3))/(V_g**(1/3) - 1)
print("T* = {0}".format(T1_p))

#P1_pp = (alfa1 / kT1) * T * V_g ** 2


V_g = 1.2466098659954163
alpha*_i = 7.228619486079665e-05
V* = 116.27535121764627
840000000000.0
P* = 5.486473153029088e-19
T* = 5686.526831659602


Componente = "1-Heptanol"

T = 323.15 K

V_g = 1.2466098659954163

alpha*_i = 7.228619486079665e-05 1 / K

V* = 116.27535121764627 cm^3 / mol

P* = 5.486473153029088e-19 J / cm^3

T* = 5686.526831659602 

"""
Al comparar los valores obtenidos con los reportados en la Tabla 6 de 
Experimental study and ERAS modeling of the excess molar enthalpy of
(acetonitrile + 1-heptanol or 1-octanol) mixtures at (298.15, 313.15, and 323.15) K
and atmospheric pressure

http://www.sciencedirect.com/science/article/pii/S0021961408000207

se reproducen los valores de volumen, temperatura, (alpha*_i) y menos la presión caracteristica

"""


In [7]:
def presion_cal(x):
    
    kT1 = x
    
    alfa1 = 9.90e-4
    alpha_p = 7.228619486079665e-05
    #kT1 = 8.40e5
    V_g = 1.2466098659954163
    
    Av1 = -5.60
    Ah1 = -18.99
    
    T = 323.15
    
    P1_p = 553.20
    
    #var_1 = ((alfa1 - alfa_p) * T * V_g ** 2) / ((kT1 - alfa_p * T * (Av1/Ah1)))
    
    var_1 = P1_p * ((kT1 - alfa_p * T * (Av1/Ah1)))
    
    f1 = P1_p * ((kT1 - alfa_p * T * (Av1/Ah1))) - ((alfa1 - alfa_p) * T * V_g ** 2)
    
    #f1 = 1 - ((alfa1 - alfa_p) * T * V_g ** 2) / ((kT1 - alfa_p * T * (Av1/Ah1))) / P1_p
    
    print(f1, var_1)
    
    return f1


sol = fsolve(presion_cal,[8.40e-4], xtol=1e-5)
print("kT = {0}".format(sol))

[-3.80687573] [-3.34601198]
[-3.80687573] [-3.34601198]
[-3.80687573] [-3.34601198]
[-3.80687572] [-3.34601198]
[ -2.50864454e-08] [ 0.46086372]
[ -1.11022302e-16] [ 0.46086374]
kT = [ 0.00772155]


En el caso $P^*_i$ con los mismos valores encontrados acá que corresponden a los repotados en la Tabla 6 de *Experimental study and ERAS modeling of the excess molar enthalpy of (acetonitrile + 1-heptanol or 1-octanol) mixtures at (298.15, 313.15, and 323.15) K and atmospheric pressure*

http://www.sciencedirect.com/science/article/pii/S0021961408000207

no reproduce el mismo valor reportado en la misma Tabla 6 y poder continuar el cálculo de los parámetros de la mezcla binaria.

Mientras se resuelve la duda anterior.

Para el caso de los parámetros de interacción binaria se puede proceder:

- Determinar los parámetros caracteristicos de cada componente puro
- Calcular las fracciones de volumen
- Suponer un valor para los parámetos: $ \Delta v^*_{12}$, $ \Delta h^*_{12}$, $ X_{12} $, $ K_{12} $
- Calcular los parámetros de la mezcla con los valores supuestos
- Determinar el volumen molar de exceso con el modelo ERAS
- Calcular la función objetivo hasta una convergencia de 1e-6 arpoximadamente: 

$$ RD = \frac{100}{N} | \sum\limits^{N}_{i=1} {\frac{V^E_{i-exp} - V^E_{i-ERAS}}{V^E_{i-exp}}}| $$




